### DISCUSSION TITLE
Allow $reset() to accept state props and not reset the whole store

### DISCUSSION THREAD

**Discussion by whitespacecode** [vuejs/pinia#1831] [Ideas]
I would like to reset just one prop in my pinia state instead of the whole store.
For now you can achieve this with this
```
//Store
import _pick from "lodash.pick";

const defaultState = {
  foo: 'bar',
  boo: 'far'
};

export const useStore = defineStore('store', {
  state: () => ({...defaultState}),
  actions: {
    reset(keys) {
      Object.assign(this, keys?.length
        ? _pick(defaultState, keys)
        : defaultState // if no keys provided, reset all
      );
    }
  }
})

//File
useStore().reset(['foo']);

```

How great would it be if the you could just do this
`store.$reset(['foo']);`

**Comment by posva** [comment]
As discussed in some issues, `$reset()` is meant to stay simple. People can add custom implementation or plugins to add this kind of features
